### PR TITLE
[SERV-326] Fix batch job status update bug

### PR DESCRIPTION
**Pull request by ksclarke**
* Update app and container dependencies (for security fixes)
* Add two new ignores/excludes to the warnings the Jar process spits out
* Include Eclipse syntax fixes to BatchJobStatusHandler and GetJobStatusesHandler
* Fix MatchingOpNotFoundHandler so it handles edge cases better (i.e., the bug)
* Add tests for MatchingOpNotFoundHandler's improved edge case handling

This PR aims to fix the bug where Bucketeer won't respond to an endpoint request when the request it receives isn't complete.

Not getting a success or failure response, our AWS Lambda (the requestor) keeps waiting until it times out. This causes it to unnecessarily run up a charge on AWS.

### CODE REVIEW

**Review by markmatney — COMMENTED** [review]
This looks to me like it works, but my main question is: if Bucketeer is the application that constructs the callback URLs, why not make it impossible to create an invalid one in the first place? Assuming that the recent errors were due to a malformed input CSV, the lack of null checks for `edu.ucla.library.bucketeer.Item.getID()`, for example, seems to me to be the root of the problem, because at the point where that method returns null or empty, we already know that problems lie ahead, and we could immediately reply to the client with an error, rather than spin up a lambda and wait for the inevitable failure. That would be a shorter feedback loop for our users.

File: src/main/java/edu/ucla/library/bucketeer/handlers/MatchingOpNotFoundHandler.java [hunk 1]
```diff
@@ -25,25 +25,63 @@
...
+                    error(aContext, HTTP.BAD_REQUEST, LOGGER.getMessage(MessageCodes.BUCKETEER_600));
+                } else if (path.matches(MISSING_JOB_RE)) {
+                    error(aContext, HTTP.BAD_REQUEST, LOGGER.getMessage(MessageCodes.BUCKETEER_601));
+                }
+            } else if (path.matches(MISSING_STATUS_RE)) { // Check that updates without a status are caught
+                error(aContext, HTTP.BAD_REQUEST, LOGGER.getMessage(MessageCodes.BUCKETEER_602));
+            } else if (HttpMethod.PATCH.name().equals(method)) { // Handle other random PATCH API requests
+                error(aContext, HTTP.BAD_REQUEST, LOGGER.getMessage(MessageCodes.BUCKETEER_162, path));
+            }
+        } // Ignore successful contexts, their responses have been returned
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/main/java/edu/ucla/library/bucketeer/handlers/MatchingOpNotFoundHandler.java:68]: I'm not sure I'm following... can you explain more? Be more specific?

**Comment by ksclarke** [comment]
Since this is a service, we cannot assume Bucketeer is the one who constructs the callback URLs. It defines the URL but anything can call that URL so we want to be defensive in our handling of the callbacks.

I agree that it would be better to fail earlier for the user from the perspective of using Bucketeer. That's what the validation ticket / work is for, though. That layer may be implemented in Item, as you suggest, or may be a layer over Bucketeer so it doesn't even get to Item. The latter seems more likely unless we abstract Item out into the validation layer, since we want to validate incoming data on all our CSV-based services (but maybe Item has a role in that).

File: src/main/java/edu/ucla/library/bucketeer/handlers/MatchingOpNotFoundHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/main/java/edu/ucla/library/bucketeer/handlers/MatchingOpNotFoundHandler.java:68]: Okay. I wasn't sure that it wasn't possible to be handling an error without the context being failed, but I see what you're saying. I'll remove.

**Review by markmatney — APPROVED** [review]
I still have some concerns about the lack of input validation, but considering the constraints that we're working under am amenable to waiting a bit longer for the more general/robust solution that we have queued up.

**Review by DRickard — APPROVED** [review]
